[Docs] Update backtracing documentation for Linux.

al45tair opened the pull request:
Update the backtracing documentation with details on static linking, frame pointer usage and a couple of other matters.

rdar://117679284

al45tair committed:
[Docs] Update backtracing documentation for Linux.

Update the backtracing documentation with details on static linking,
frame pointer usage and a couple of other matters.

rdar://117679284

al45tair commented:
@swift-ci Please smoke test

mikeash commented on docs/Backtracing.rst:
```diff
@@ -229,3 +259,57 @@ Other Darwin (iOS, tvOS)
+the following locations::
+
+    <swift-root>/libexec/swift/<platform>
+    <swift-root>/libexec/swift/<platform>/<arch>
```
Will people know the exact form that `<platform>` and `<arch>` will have? If not, would be good to briefly describe that.

al45tair replied:
I've added a description.

al45tair committed:
[Docs] Added a description of the platform and arch path components.

Describe what `<platform>` and `<arch>` mean in the search paths
for the `swift-backtrace` binary.

rdar://117679284

al45tair commented:
@swift-ci Please smoke test and merge.
